### PR TITLE
Disable install button for all add-ons on Fenix

### LINKED ISSUE #9864: Enable installation buttons for supported extensions in Fenix on dev
We're close to being able to install Fenix add-ons from AMO, so we should enable the installation button for those add-ons that are compatible (Recommended for Android). This change should be behind a flag so we can turn it on when the Fenix work is ready. I'm thinking we should enable it on dev and stage when it lands, and prod should wait until shortly before the Fenix changes make it to Release.

For QA:
I haven't enabled this flag on dev yet, as I wanted you to be able to verify the fix for #9867 first, and that would be difficult to do if this change landed enabled. I think I'll change the milestone for this to the next one, and then enable the flag on dev for this at the next tag.

### LINKED ISSUE #9867: "Add to Firefox" button is currently enabled in Fenix
There's an [issue to enable the "Add to Firefox" button in Fenix here](https://github.com/mozilla/addons-frontend/issues/9864). However that should only be turned on when Fenix has been updated to handle installs from AMO for recommended. 

The Fenix team noticed that the "Add to Firefox" button is enabled today.

IMG_7457

It should be disabled until Fenix is ready.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #9867 
Fixes #9864

### CODE REVIEW

File: src/core/utils/compatibility.js [hunk 1]
```diff
@@ -166,6 +166,13 @@ export function isCompatibleWithUserAgent({
     return { compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX };
   }
 
+  // Fenix does not support add-ons (yet?).
+  // See: https://github.com/mozilla-mobile/fenix/issues/1134
+  // See also: https://github.com/mozilla/addons-frontend/issues/7963
```
Comment (eviljeff) [inline src/core/utils/compatibility.js:171]: https://github.com/mozilla/addons-frontend/issues/9864 is probably a better issue to mention?

File: src/core/utils/compatibility.js [hunk 2]
```diff
@@ -166,6 +166,13 @@ export function isCompatibleWithUserAgent({
     return { compatible: false, reason: INCOMPATIBLE_NOT_FIREFOX };
   }
 
+  // Fenix does not support add-ons (yet?).
```
Comment (eviljeff) [inline src/core/utils/compatibility.js:169]: mmmmmmmmmm, that's not really true.

File: src/core/utils/compatibility.js [hunk 3]
```diff
@@ -166,6 +166,13 @@ export function isCompatibleWithUserAgent({
...
+  // See: https://github.com/mozilla-mobile/fenix/issues/1134
+  // See also: https://github.com/mozilla/addons-frontend/issues/7963
+  if (isFirefoxForAndroid(userAgentInfo)) {
+    return { compatible: false, reason: INCOMPATIBLE_ANDROID_UNSUPPORTED };
```
Comment (eviljeff) [inline src/core/utils/compatibility.js:173]: so this replaces the more nuanced logic below that checks the OS and the compatibility by just making all add-ons incompatible with Fennec on Android regardless of the compatibility status?

File: src/core/utils/compatibility.js (hunk 1 above)
Comment (bobsilverberg) [reply src/core/utils/compatibility.js:171]: Yeah, that was a copy and paste of the code that was there before the patch that removed it. I'll update it.

File: src/core/utils/compatibility.js (hunk 2 above)
Comment (bobsilverberg) [reply src/core/utils/compatibility.js:169]: Yeah, we should change it to say add-ons are not installable from AMO on Fenix yet.

File: src/core/utils/compatibility.js (hunk 3 above)
Comment (bobsilverberg) [reply src/core/utils/compatibility.js:173]: That is correct. For now, we want all add-ons to appear incompatible if someone is running Firefox for Android. There's an issue to change this, when they do become installable, at which point we can being some of the logic back.

Comment (willdurand) [reply src/core/utils/compatibility.js:173]: couldn't we use a feature flag instead then?

Comment (bobsilverberg) [reply src/core/utils/compatibility.js:173]: Yeah, I can probably get that done in time for the tag. This was an attempt at a quick fix for a bug that is currently present in production. I didn't know if we were going to have to hotfix it or not, so I just wanted a patch that fixes it.

Now that I know we can wait until Thursday for this fix, we can probably address #9864 in this patch as well.

**Comment by bobsilverberg** [comment]
@eviljeff I have addressed all of the comments, and also added a flag to address #9864. I also did a bit of extra cleanup in the test helpers and tests as we no longer differentiate between Fennec and Fenix, so we shouldn't have anything directly referencing Fenix.

File: tests/unit/core/utils/test_compatibility.js [hunk 4]
```diff
@@ -1115,12 +1127,6 @@ describe(__filename, () => {
   });
 
   describe('isFirefoxForIOS', () => {
-    it('returns false for Firefox Fenix', () => {
```
Comment (eviljeff) [inline tests/unit/core/utils/test_compatibility.js:1118]: 😕 FirefoxForIOS was testing for Fenix but now isn't?

**Review by eviljeff — COMMENTED** [review]
I don't understand the changes in test_compatablity and /helpers.js - fenix is now fennec??  Where did that happen in the changes in the live code?

**Comment by bobsilverberg** [comment]
> I don't understand the changes in test_compatablity and /helpers.js - fenix is now fennec?? Where did that happen in the changes in the live code?

Both Fenix and Fennec are now Firefox for Android. A lot of the changes regarding that were actually made in earlier patches, but I neglected to fix up the tests. Because we are assuming that anyone on an Android browser is using Fenix, there were tests added specifically for Fenix, but that doesn't really make sense, as nothing in the code base differentiates between Fennec and Fenix anymore.

This is why I combined both sets of user agents into `android` in `tests/unit/helpers.js`, and also why I removed any test cases that were specifically for Fenix. We don't need any of those anymore. Instead we just need a test case for Firefox for Android, and we already had those in place.

File: src/core/utils/compatibility.js [hunk 5]
```diff
@@ -166,6 +169,18 @@ export function isCompatibleWithUserAgent({
...
   }
 
+  if (isFirefoxForAndroid(userAgentInfo)) {
+    // Add-ons are not installable in Fenix from AMO yet.
+    // See: https://github.com/mozilla/addons-frontend/issues/9864
+    if (!_config.get('enableFeatureAllowAndroidInstall')) {
+      return { compatible: false, reason: INCOMPATIBLE_ANDROID_UNSUPPORTED };
+    }
+    // We need to check that the add-on is installable.
+    if (!isAndroidInstallable({ addon })) {
```
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply src/core/utils/compatibility.js:179]: Hmm, I suppose we should still be checking that. It's annoying that there are two sets of criteria to specify that an extension can be installed on Android (`currentVersion.compatibility[CLIENT_APP_ANDROID]` and `promoted`), but I guess both are still valid. I'll add that back in.

File: tests/unit/core/utils/test_compatibility.js (hunk 4 above)
Comment (bobsilverberg) [reply tests/unit/core/utils/test_compatibility.js:1118]: As discussed below, any tests that existed for Fenix have now been combined into tests for Firefox for Android.

File: src/core/utils/compatibility.js (hunk 5 above)
[the remark this answers is not in the document]
Comment (bobsilverberg) [reply src/core/utils/compatibility.js]: I added this back in. I accidentally did an `amend` commit so it doesn't show up in a separate commit.

**Review by eviljeff — APPROVED** [review]

**Pull request merged**